usersync/encoder.go

```diff
@@ -0,0 +1,19 @@
+package usersync
+
+import (
+	"encoding/base64"
+	"encoding/json"
+)
+
+type Encoder interface {
+	Encode(c *Cookie) string // Encode a cookie into a base 64 string
+}
+
+type EncoderV1 struct{}
```

Comment:
Are we planning to implement other encoders? If so, are those also `base64` encoders? In either scenario: I believe we still should rename. Maybe:
```
12 -  type EncoderV1 struct{}
   +  type Base64Encoder struct{}
usersync/cookie.go
```
Or
```
12 -  type EncoderV1 struct{}
   +  type Base64EncoderV1 struct{}
usersync/cookie.go
```

Reply:
I'd assume there are plans for other encoders in the future, I'm not exactly sure what those would look like, @SyntaxNode was the one who suggested to have a V1 attached to the encoder instance. But for this PR, this will be the only encoder implemented.